src/neo/SmartContract/Native/Tokens/NeoToken.cs

```diff
@@ -43,225 +31,48 @@ protected override void OnBalanceChanging(ApplicationEngine engine, UInt160 acco
             if (amount.IsZero) return;
             if (state.VoteTo != null)
             {
-                engine.Snapshot.Storages.GetAndChange(CreateStorageKey(Prefix_Candidate).Add(state.VoteTo)).GetInteroperable<CandidateState>().Votes += amount;
+                var candidate_state = engine.Snapshot.Storages.GetAndChange(CreateStorageKey(Prefix_Candidate).Add(state.VoteTo)).GetInteroperable<CandidateState>();
+                candidate_state.Votes += amount;
                 engine.Snapshot.Storages.GetAndChange(CreateStorageKey(Prefix_VotersCount)).Add(amount);
-            }
-        }
 
-        private void DistributeGas(ApplicationEngine engine, UInt160 account, NeoAccountState state)
-        {
-            BigInteger gas = CalculateBonus(state.Balance, state.BalanceHeight, engine.Snapshot.PersistingBlock.Index);
-            state.BalanceHeight = engine.Snapshot.PersistingBlock.Index;
-            GAS.Mint(engine, account, gas);
-        }
-
-        private BigInteger CalculateBonus(BigInteger value, uint start, uint end)
-        {
-            if (value.IsZero || start >= end) return BigInteger.Zero;
-            if (value.Sign < 0) throw new ArgumentOutOfRangeException(nameof(value));
-            BigInteger amount = BigInteger.Zero;
-            uint ustart = start / Blockchain.DecrementInterval;
-            if (ustart < Blockchain.GenerationAmount.Length)
-            {
-                uint istart = start % Blockchain.DecrementInterval;
-                uint uend = end / Blockchain.DecrementInterval;
-                uint iend = end % Blockchain.DecrementInterval;
-                if (uend >= Blockchain.GenerationAmount.Length)
+                if (candidate_state.Votes == 0)
                 {
-                    uend = (uint)Blockchain.GenerationAmount.Length;
-                    iend = 0;
+                    UInt160 voteeAddr = Contract.CreateSignatureContract(state.VoteTo).ScriptHash;
+                    foreach (var (key, _) in engine.Snapshot.Storages.Find(CreateStorageKey(Prefix_VoterRewardPerCommittee).Add(voteeAddr).ToArray()))
```

Comment:
Use the public key as the storage key instead of `voteeAddr`.

Reply:
Done in https://github.com/neo-project/neo/pull/1716/commits/5ece87f788bfd3a56e3b137643278a2692d00ecd